packages/cli/src/commonlib/telemetry.ts

```diff
@@ -102,4 +99,37 @@ export class CliTelemetryReporter implements TelemetryReporter {
   async flush(): Promise<void> {
     await this.reporter.flush();
   }
+
+  addCommonProperties(rootPath: string | undefined, properties: { [p: string]: string }) {
+    const projectSettings = getProjectSettingsForCommonTelemetry(rootPath);
+
+    if (
+      properties[TelemetryProperty.ProjectId] === "unknown" ||
+      properties[TelemetryProperty.ProjectId] === undefined
+    ) {
+      properties[TelemetryProperty.ProjectId] = projectSettings?.projectId
+        ? projectSettings?.projectId
+        : "unknown";
+    }
+
+    if (!properties[TelemetryProperty.IsFromSample]) {
+      properties[TelemetryProperty.IsFromSample] = projectSettings?.isFromSample;
+    }
+
+    if (!properties[TelemetryProperty.ProjectSettingsVersion]) {
+      properties[TelemetryProperty.ProjectSettingsVersion] = projectSettings?.version;
+    }
+
+    if (!properties[TelemetryProperty.SolutionSettings]) {
+      properties[TelemetryProperty.SolutionSettings] = projectSettings?.solutionSettings;
```

Comment:
do we really need to record the whole solution settings and plugin settings? The content of them may be out of control and cause problems.